pkg/operation/botanist/controlplane.go

```diff
@@ -278,6 +278,12 @@ func (b *Botanist) deployOrRestoreControlPlane(ctx context.Context, controlPlane
 	return controlPlane.Deploy(ctx)
 }
 
+// RestoreControlPlane restores the ControlPlane custom resource (purpose normal)
+func (b *Botanist) RestoreControlPlane(ctx context.Context) error {
+	b.Shoot.Components.Extensions.ControlPlane.SetInfrastructureProviderStatus(b.Shoot.Components.Extensions.Infrastructure.ProviderStatus())
+	return b.Shoot.Components.Extensions.ControlPlane.Restore(ctx, b.GetShootState())
+}
```

Comment:
Why is this function needed? It seems to do the same as in `DeployControlPlane`, right?

Reply:
DeployControlPlane will either deploy or restore the controlplane depending on whether isRestorePhase() is true, but now we also need to call Restore() during the migration phase (where it is not necessary to check if only deployment is needed). 

I thought that a separate method would be a bit more clear.

Reply:
OK, thanks.